Modules/mathmodule.c

```diff
@@ -2471,14 +2474,60 @@ vector_norm(Py_ssize_t n, double *vec, double max, int found_nan)
             x = vec[i];
             assert(Py_IS_FINITE(x) && fabs(x) <= max);
             x *= scale;
-            x = x*x;
             assert(x <= 1.0);
+
+            t = x * T27;
+            hi = t - (t - x);
+            lo = x - hi;
+            assert (hi + lo == x);
+
+            x = hi * hi;
+            assert(csum >= x);
+            oldcsum = csum;
+            csum += x;
+            frac += (oldcsum - csum) + x;
+
+            x = 2.0 * hi * lo;
+            assert(csum >= x);
+            oldcsum = csum;
+            csum += x;
+            frac += (oldcsum - csum) + x;
+
+            x = lo * lo;
             assert(csum >= x);
             oldcsum = csum;
             csum += x;
             frac += (oldcsum - csum) + x;
         }
-        return sqrt(csum - 1.0 + frac) / scale;
+        result = sqrt(csum - 1.0 + frac);
+
+        x = result;
+        t = x * T27;
+        hi = t - (t - x);
+        lo = x - hi;
+        assert (hi + lo == x);
+
+        x = -lo * lo;
+        assert(csum >= x);
+        oldcsum = csum;
+        csum += x;
+        frac += (oldcsum - csum) + x;
+
+        x = -2.0 * hi * lo;
+        assert(csum >= x);
```

Comment:
Same comment as for the similar assert above (`x` may be positive or negative here).

Reply:
Same as above